coffee-chats/src/main/webapp/src/util/fetch.ts

```diff
@@ -26,7 +29,31 @@ export function useFetch(url: string): any {
         setData(json);
       }
     })();
-  }, [url]);
+  }, [url, forceUpdate]);
 
-  return data;
+  return [data, () => {
+    setForceUpdate(forceUpdate + 1);
+  }];
+}
+
+/**
+ * Sends a POST request to the specified URL, passing specified data as parameters
+ *
+ * @param url: URL to send the POST request to
+ * @param data: Data to pass as urlencoded parameters
+ */
+export async function postData(url: string, data: Map<string, string>) {
+  const requestBody = new URLSearchParams();
+
+  data.forEach((value, key) => {
+    requestBody.append(key, value);
+  });
+
+  await fetch(url, {
+    "method": "POST",
+    body: requestBody,
+    headers: {
+      "Content-Type": "application/x-www-form-urlencoded"
+    }
```

Comment:
I think it would be sensible to add some basic check to see if the post was successful and return this to the calling function. As a generic function you may want to only perform certain UI actions if the POST request was successful.